core/src/main/kotlin/xtdb/arrow/ListExpression.kt

```diff
@@ -0,0 +1,10 @@
+package xtdb.arrow
+
+import xtdb.vector.IVectorWriter
+import org.apache.arrow.vector.types.pojo.Field
+
+interface ListExpression {
+    fun field(): Field
```

Comment:
in Kotlin this would be `val field: Field` and `val size: Int`, called from Clojure as `getField` and `getSize` respectively